portal-bridge/src/types/state/utils.rs

```diff
@@ -0,0 +1,52 @@
+use alloy_primitives::{keccak256, Address, B256};
+
+pub fn full_nibble_path_to_address_hash(key_path: &[u8]) -> B256 {
```

Comment:
nit: Most (I would say **all not-trivial**) public functions should have documentation. This also "fixes" the confusion with having more generic names. This greatly helps with coding because of the IDE.

In this case it is more acceptable to keep your name because we have `address_to_nibble_path` next to it.